overlord/hookstate/ctlcmd/get.go

```diff
@@ -354,34 +356,53 @@ func (c *getCommand) getInterfaceSetting(context *hookstate.Context, plugOrSlot
 	})
 }
 
-func (c *getCommand) getRegistryView(ctx *hookstate.Context, plugName string) error {
+func (c *getCommand) getRegistryValues(ctx *hookstate.Context, plugName string, requests []string) error {
 	if c.ForcePlugSide || c.ForceSlotSide {
 		return fmt.Errorf(i18n.G("cannot use --plug or --slot with --view"))
 	}
-
 	ctx.Lock()
 	defer ctx.Unlock()
+
+	reg, view, err := getRegistryView(ctx, plugName)
+	if err != nil {
+		return fmt.Errorf("cannot get registry: %v", err)
+	}
+
+	tx, err := registrystate.RegistryTransaction(ctx, reg)
+	if err != nil {
+		return err
+	}
+
+	res, err := registrystate.GetViaViewInTx(tx, view, requests)
+	if err != nil {
+		return err
+	}
+
+	return c.printPatch(res)
+}
+
+func getRegistryView(ctx *hookstate.Context, plugName string) (*registry.Registry, *registry.View, error) {
```

Comment:
would it make sense to have a method on View that lets access the corresponding Registry to simplify things a bit here?